File: frontend/src/app/models/entity/estructura-roles-usuarios.ts
import { RolesUsuarios } from './roles-usuario';

export class EstructuraRolesUsuarios {
    constructor (
        public  servidor  ?: string,
        public  rolid ?: number,
        public  hdgcodigo ?: number,
        public  esacodigo ?: number,
        public  cmecodigo ?: number,
        public  codigorol ?: string,
        public  nombrerol ?: string,
        public  descripcionrol ?: string,
        public  idusuario    ?:number,
        public  detalle  ?: RolesUsuarios[],
        ) {
            
        }
}

File: backend/Modulos/models/estructuraRolesUsuarios.go
package models

// EstructuraRolesUsuarios is...
type EstructuraRolesUsuarios struct {
	SERVIDOR       string          `json:"servidor"`
	IDROL          int             `json:"rolid"`
	HDGCODIGO      int             `json:"hdgcodigo"`
	ESACODIGO      int             `json:"esacodigo"`
	CMECODIGO      int             `json:"cmecodigo"`
	CODIGOROL      string          `json:"codigorol"`
	NOMBREROL      string          `json:"nombrerol"`
	DESCRIPCIONROL string          `json:"descripcionrol"`
	IDUSUARIO      int             `json:"idusuario"`
	ACCION         string          `json:"accion"`
	DETALLE        []RolesUsuarios `json:"detalle"`
}
